### DISCUSSION TITLE
Questions about video clips

### DISCUSSION THREAD

**Discussion by ohjohnsen** [photoprism/photoprism#1081] [Q&A] [answered]
Hiya.  Thanks for all the awesome work with PhotoPrism so far!

I've got a couple of questions regarding video clips:

1.  It seems that some of my clips aren't possible to play in the frontend.  I just get a spinner when trying to play them, which I guess means it's beginning to transcode the files in the backend.  Is it possible to have PhotoPrism do all the transcoding beforehand?  I don't mind using the drive space for the files.  I'd much rather have a snappy frontend than save a few GB. :-)  Also, where are the transcoded files stored?

2.  Some time ago I converted some of the clips into more compressed formats, probably to send them to family members over the Internet or something.  I any case, those duplicates does not have the metadata from the original files and are therefore set to the date on which I did the duplication.  That's fine..  But now I want to remove those duplicates.  What's the best process on doing this so that both the video image file and any sidecar files are also removed?  Is just archiving the video and the video image file and then deleting the archived files enough?

**Comment by lastzero** [comment, marked as answer]
See #703 - you may need to transcode them, which either happens "on the fly" when you play it. This may be too slow for large files. Please run the "photoprism convert" command in a terminal in this case to pre-transcode all videos.

**Reply by ohjohnsen** [reply]
Is there an option to set automatic pre-transcoding when importing new files?  Or do I have to run this command after each import containing video files?

**Reply by lastzero** [reply]
Not yet as pre-transcoding is a new feature - I'd rather not run this automatically on every import until we gathered more feedback. Hardware transcoding for example may be unstable, also for reasons outside our control.

**Reply by ohjohnsen** [reply]
Ok, thanks.  It's currently pre-transcoding the video files in my library, so I'll see how it goes.  I think this is a very nice feature.

Out of curiosity, which timezone are you in?  Did not expect to get an answer from you after two minutes half past seven on a Sunday morning (I'm in Norway - CET timezone). :-)

**Reply by lastzero** [reply]
Same time zone, Berlin. Got up early! :coffee:

**Reply by ohjohnsen** [reply]
Would definitely be asleep myself if my 16 month old daughter wasn't adamant about getting up at 5 o'clock. 😁  Well, good morning to you then, and enjoy your ☕.

**Reply by streiman** [reply]
I tried

sudo docker-compose photoprism convert

but there seems to be no option "convert". Do I miss something?

**Reply by ohjohnsen** [reply]
@streiman It doesn't work like that.  You have to shell into the running PhotoPrism container.  Run this command on your computer running the PhotoPrism container:  `docker exec -it photoprism bash`

Note that the "photoprism" argument is the name of the PhotoPrism container.  On my unRAID server the container is named "photoprism".  If you have named it something else, then you have to use that as the argument instead.

**Reply by streiman** [reply]
Thank you. I really have to read a little bit about docker, I guess. But now it is converting.

**Reply by streiman** [reply]
But one more question: Is there a possbility to run that "photoprism convert" from a cron job on the computer running the PhotoPrism container? I want to trigger a workflow where new photos are copied to my original files directory according to my archive schema, trigger the indexing and then running the convert (for the case that there are new videos uploaded).

**Reply by ohjohnsen** [reply]
Sure.  Just run `docker exec -it photoprism photoprism convert` instead (you need double "photoprism" arguments, as the first is the container name and the second is the command inside the container).

**Reply by streiman** [reply]
I start to understand 🙃 thank you for your help.

**Reply by ohjohnsen** [reply]
Docker isn't that complicated really, when you get down to it.  A Docker container is just a Linux virtual machine/computer.  You use a Docker image to create a Docker container, and a image is a read-only file that contains source code, libraries, dependencies and stuff you need to run what you want.  If you really wanted to you could have a completely empty Docker image, and when spinning up a container from that image would be similar to having a clean Linux installation without any extra packages.

The creator of an image has a Dockerfile that's just a recipe for what a Docker image should contain.  Here's the Dockerfile for PhotoPrism for example: https://github.com/photoprism/photoprism/blob/develop/docker/photoprism/Dockerfile   As you can see the file specifies a build action, a bunch of package installations, some file handling, and finally the PhotoPrism server being started.

**Reply by acranox** [reply]
I'd also love to see transcoding on import be an option.  I just ran `photoprism convert` on my existing library, which took days to run.  I'm running on a low-power system, so on-the-fly transcoding performance isn't great.  I then re-ran convert (to catch any new movies I imported), which appears to have crawled through the whole library again, and it still took 45 minutes (with 0 new movies transcoded), and caused a massive amount of disk activity in the process.  So doing a nightly convert job to catch newly imported files isn't really feasible for me.  An option to just transcode new videos when they are imported would be great for my use case.

For reference, my library is about 60,000 photos, 6,000 videos, totaling over 200GB

**Reply by streiman** [reply]
An pre-transcoding on import really would be great or some options for photoprism convert like "only work on files added today / added since ..."

But is it possible to limit convert on a folder like I can reindexing in the ui? That might be a kind of workaround to run it only where my script put new files.

**Reply by lastzero** [reply]
Scope of the convert command can now be limited (new preview build soon available for testing): https://github.com/photoprism/photoprism/commit/cd63d33bc19950efdb8ca8886a2377a4f708b655

**Reply by streiman** [reply]
Works, thank you very much.
